### PR TITLE
refactor(publikator): implement new Discussion/Comment interface

### PR DESCRIPTION
**Pull request by patrickvenetz**
This Pull Request uses refactor Discussion/Comment interface to provide memos in publikator.

To hide unnecessary vote buttons, https://github.com/republik/plattform/pull/3/commits/2297a7e149ee2d2aaa4f49b9a10592c6e6ce6b75 checks for defined votes handlers.

Bildschirmfoto 2022-02-01 um 17 51 39

### CODE REVIEW

File: packages/styleguide/src/components/Discussion/Tree/CommentNode.tsx [hunk 1]
```diff
@@ -270,6 +270,15 @@ const CommentNode = ({
...
 
   if (isExpanded) {
+    const { handleUpVote, handleDownVote, handleUnVote } = actions
+    const voteActions = handleUpVote &&
+      handleDownVote &&
+      handleUnVote && {
+        handleUpVote,
+        handleDownVote,
+        handleUnVote
+      }
```
Comment (trm217) [inline packages/styleguide/src/components/Discussion/Tree/CommentNode.tsx:280]: I am not sure if this is the best way to handle this. TypeScript is requiring all these handlers to be present. I suppose the best way would be to make them optional on the CommentNode and handle it with that. (With this implementation, the Project could no longer be built if TS-Errors block the build)

**Review by trm217 — APPROVED** [review]
Looks good for now, however we should get back and add a CommentNode version without VoteButtons.

File: apps/publikator/components/editor/modules/memo/MemoTree.js [hunk 2]
```diff
@@ -1,116 +1,178 @@
...
+          onClick: () => setIsEditing(true)
         },
-        error => ({ error })
-      ),
-    unpublishComment: comment =>
-      unpublish(comment.id).then(
-        async response => {
-          !!onUnpublished && (await onUnpublished(response.data.memo))
-          return { ok: true }
+        !!userCanEdit && {
```
[the remark this answers is not in the document]
Comment (patrickvenetz) [reply apps/publikator/components/editor/modules/memo/MemoTree.js:88]: Indeed, there is no "admin"-unpublish required.

File: apps/publikator/components/editor/modules/memo/MemoTree.js [hunk 3]
```diff
@@ -1,116 +1,178 @@
...
+  return (
+    <DiscussionContext.Provider value={discussionContext}>
+      <Container memo={rootMemo} />
+    </DiscussionContext.Provider>
+  )
+}
+
+export default compose(withT, withMe, withMemos)(Tree)
+
+const Container = ({ memo }) => {
```
[the remark this answers is not in the document]
Comment (patrickvenetz) [reply apps/publikator/components/editor/modules/memo/MemoTree.js:73]: God damn it: https://github.com/republik/plattform/pull/3/commits/92a629c42e070ede44da5048e13ccc3c437018b4

File: apps/publikator/lib/translations.json [hunk 4]
```diff
@@ -1202,7 +1202,31 @@
     },
     {
       "key": "memo/modal/warning/newDoc",
-      "value": "Sie können erst Kommentare hinterlassen, nachdem Sie den Beitrag kommittet haben."
+      "value": "Sie können eine Notiz hinterlassen, nachdem Sie den Beitrag comitted haben."
```
Comment (annatraussnig) [inline apps/publikator/lib/translations.json:1205]: `comitted` with one `m`?

File: packages/styleguide/src/components/Discussion/Tree/CommentNode.tsx (hunk 1 above)
Comment (annatraussnig) [reply packages/styleguide/src/components/Discussion/Tree/CommentNode.tsx:280]: +1 
The above fix is a bit confusing – plus then the type structure doesn't reflect the comments' reality anymore. How complex would it be to nest vote actions in actions and make them optional as a whole, but individually mandatory if present?
```
type Actions = {
  ...
  voteActions?: {
     handleUpVote: Fn
     handleDownVote: Fn
     handleUnVote: Fn
  }
}
```

File: apps/publikator/components/editor/modules/memo/MemoTree.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (annatraussnig) [reply apps/publikator/components/editor/modules/memo/MemoTree.js:73]: I for one don't mind at all 😇 
It's a module called `memo`, in a file called `MemoTree`.

File: apps/publikator/components/editor/modules/memo/MemoTree.js [hunk 5]
```diff
@@ -1,116 +1,178 @@
...
+          publish(repoId, parentId, text).then(
+            async response => {
+              !!onPublished && (await onPublished(response.data.memo))
+              return { ok: true }
+            },
+            error => ({ error })
+          ),
+        editMemo: (id, text) =>
+          edit(id, text).then(
+            () => ({ ok: true }),
```
Comment (annatraussnig) [inline apps/publikator/components/editor/modules/memo/MemoTree.js:49]: do we use this `{ ok: true }` object somewhere?

**Review by annatraussnig — APPROVED** [review]

File: apps/publikator/lib/translations.json (hunk 4 above)
Comment (patrickvenetz) [reply apps/publikator/lib/translations.json]: That's a typo. Adressed in https://github.com/republik/plattform/pull/3/commits/5d1529ed28387d9a766752ec5b86b7ad1f0bc5fe

File: apps/publikator/components/editor/modules/memo/MemoTree.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (patrickvenetz) [reply apps/publikator/components/editor/modules/memo/MemoTree.js]: Reverted to prefixed Memo in https://github.com/republik/plattform/pull/3/commits/6247796aca1e289cc52751c8b9f1d48c4751a362 – doesn't hurt me and if it helps, I say that is a go!

File: packages/styleguide/src/components/Discussion/Tree/CommentNode.tsx (hunk 1 above)
Comment (patrickvenetz) [reply packages/styleguide/src/components/Discussion/Tree/CommentNode.tsx]: Addressed in https://github.com/republik/plattform/pull/3/commits/6f4a604684f2f5db75920aa8ea5c58d803307378 and going for dedicated `voteActions`.

File: apps/publikator/components/editor/modules/memo/MemoTree.js (hunk 5 above)
Comment (patrickvenetz) [reply apps/publikator/components/editor/modules/memo/MemoTree.js:49]: For submit and edit mutation, [yes](https://github.com/republik/plattform/blob/publikator-refactor-memos/packages/styleguide/src/components/Discussion/Composer/CommentComposer.js#L247-L248): `CommentComposer` calls handler (`onSubmit`) and then processes response – ok or error – to handle post composer matters.

For unpublish, no. But kept it, for sanity purposes, the same.

**Pull request merged**